pkg/generators/openapi.go

```diff
@@ -553,19 +557,27 @@ func (g openAPITypeWriter) validatePatchTags(m *types.Member, parent *types.Type
 	return nil
 }
 
-func defaultFromComments(comments []string) (interface{}, error) {
+func defaultFromComments(comments []string) (interface{}, *types.Name, error) {
 	tag, err := getSingleTagsValue(comments, tagDefault)
 	if tag == "" {
-		return nil, err
+		return nil, nil, err
 	}
 	var i interface{}
-	if err := json.Unmarshal([]byte(tag), &i); err != nil {
-		return nil, fmt.Errorf("failed to unmarshal default: %v", err)
+	if id, ok := parseAsRef(tag); ok {
+		parsedName := types.ParseFullyQualifiedName(id)
+		return nil, &parsedName, nil
+	} else if err := json.Unmarshal([]byte(tag), &i); err != nil {
+		return nil, nil, fmt.Errorf("failed to unmarshal default: %v", err)
 	}
-	return i, nil
+	return i, nil, nil
 }
 
 func mustEnforceDefault(t *types.Type, omitEmpty bool) (interface{}, error) {
+	// If t implements custom JSON marshalling, all of this logic is likely wrong
```

Comment:
yeah, I think that's fine.

Reply:
I think it is related to https://github.com/kubernetes/kubernetes/issues/118771

Reply:
Agreed, sorry, what I meant was "Yeah, that looks like a reasonable way to address some of the things you've been talking about". Yeah, I agree with you.

Reply:
Should we disregard all defaults for these types, including the ones in comments?

Reply:
I will open a separate Pr for this change to keep things orthogonal